### PR TITLE
openrct2: 0.2.3 -> 0.2.4

### PR DESCRIPTION
**Pull request by lightbulbjim**
###### Motivation for this change

Updating to latest version.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @geistesk

### CODE REVIEW

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
executables seem to work
leaf package

```
https://github.com/NixOS/nixpkgs/pull/72212
1 package were build:
openrct2
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build openrct2

**Review by c0bw3b — COMMENTED** [review]
`cmake` and `pkgconfig` should rather be in `nativeBuildInputs`

**Comment by lightbulbjim** [comment]
> `cmake` and `pkgconfig` should rather be in `nativeBuildInputs`

Fair enough. Updated.

File: pkgs/games/openrct2/default.nix
Comment (geistesk) [inline pkgs/games/openrct2/default.nix]: The CMakeLists.txt for the 0.2.4-release requires version 0.1.2c of the title sequences.

File: pkgs/games/openrct2/default.nix
Comment (geistesk) [inline pkgs/games/openrct2/default.nix]: The 0.2.4-release introduced a [replay](https://github.com/OpenRCT2/replays) module in its CMakeLists.txt. This sub-project should be added like the objects and title sequences.

**Review by oxzi — CHANGES_REQUESTED** [review]
Thanks for your PR. Please check the upstream CMakeLists.txt for updates, as already commented.

File: pkgs/games/openrct2/default.nix
Comment (lightbulbjim) [reply pkgs/games/openrct2/default.nix]: Oops, missed that one. Fixed, thanks!

File: pkgs/games/openrct2/default.nix
Comment (lightbulbjim) [reply pkgs/games/openrct2/default.nix]: `replays` is only used for tests, which we don't currently run. 

If we want to run tests then we'll need to include [googletest](https://github.com/google/googletest/) in the derivation. I'm not sure if there's a quick and easy way to disable the download, so we might need to patch `test/tests/CMakeLists.txt` as well.

**Comment by lightbulbjim** [comment]
Is there anything I can do to assist moving this forward?

**Comment by c0bw3b** [comment]
If we stick to not running tests, could you add an explicite cmake flag with a comment :

```nix
  cmakeFlags = [
    "-DCMAKE_BUILD_TYPE=RELWITHDEBINFO"
    "-DDOWNLOAD_OBJECTS=OFF"
    "-DDOWNLOAD_TITLE_SEQUENCES=OFF"
    "-DWITH_TESTS=OFF" # requires GTest and fetching replays
  ];
```

---

OTOH you should be able to use `gtest` from nixpkgs by passing another cmake flag `-DSYSTEM_GTEST` (look at ceph package for example) and fetch replays like other modules.
But you would need to check locally if some other problems arise by running tests, the time it takes, etc.

**Comment by lightbulbjim** [comment]
> If we stick to not running tests, could you add an explicite cmake flag with a comment :
> 
> ```nix
>   cmakeFlags = [
>     "-DCMAKE_BUILD_TYPE=RELWITHDEBINFO"
>     "-DDOWNLOAD_OBJECTS=OFF"
>     "-DDOWNLOAD_TITLE_SEQUENCES=OFF"
>     "-DWITH_TESTS=OFF" # requires GTest and fetching replays
>   ];
> ```
> 
> OTOH you should be able to use `gtest` from nixpkgs by passing another cmake flag `-DSYSTEM_GTEST` (look at ceph package for example) and fetch replays like other modules.
> But you would need to check locally if some other problems arise by running tests, the time it takes, etc.

Sure, I'm happy to try and add the tests if that's what we want to do. My intuition would have been to leave the tests out since they weren't included in the derivation previously. I guess ultimately the decision is up to @geistesk as the maintainer.

**Review by oxzi — APPROVED** [review]
Excuse my late reply. I have tested the submitted PR successfully.

Regarding the test discussion, I would like to note that by default `WITH_TESTS` is `OFF`. So I do not think this cmake flag needs to be added additionally. If you want to add the tests, I am certainly not stopping you. Anyway, the PR looks good for me in the current state.

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
leaf package

```
[5 built, 5 copied (2.6 MiB), 0.3 MiB DL]
https://github.com/NixOS/nixpkgs/pull/72212
1 package were build:
openrct2
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build openrct2

**Review by c0bw3b — APPROVED** [review]

**Pull request merged**
